sanic/app.py

```diff
@@ -637,14 +637,19 @@ def url_for(self, view_name: str, **kwargs):
 
         if external:
             if not scheme:
-                if ":" in netloc[:8]:
+                if netloc and ":" in netloc[:8]:
                     scheme = netloc[:8].split(":", 1)[0]
                 else:
                     scheme = "http"
 
-            if "://" in netloc[:8]:
+            if netloc and "://" in netloc[:8]:
                 netloc = netloc.split("://", 1)[-1]
 
+            if not netloc:
+                scheme = ""
+                external = False
```

Comment:
the branching here is getting a bit complicated, I'd maybe suggest something like
```
scheme = re.match(
    r"(?P<scheme>https?)?(?(scheme)://)(?P<netloc>\w+)",
    kwargs.pop("_scheme", "")
).groupdict()
if scheme:
    scheme, netloc = scheme['scheme'], scheme['netloc']
else:
    scheme, netloc = ""
```
but we can always address this seperately